packages/nimble-components/src/rich-text/specs/README.md

```diff
@@ -200,6 +381,69 @@ _CSS Classes and CSS Custom Properties that affect the component_
 
 _Note_: This initial component design serves as a starting point for implementation, and it may undergo changes once the visual design is completed.
 
+#### User mention element:
+
+This element serves as a container for the mapping elements and additionally functions as a configuration element. It allows the transmission of
+arbitrary values that are specific to the `@mention` users within the rich text components. Currently, this component do not have any
+configuration properties/attributes for user mentions and plays the role of containing element for the user mapping elements.
+
+_Component Name_
+
+-   `nimble-rich-text-mention-users`
+
+_Content_
+
+-   One or more `nimble-mapping-mention-user` elements
+
+#### Mapping element (user mention):
+
+This mapping element is employed to establish a connection between the value displayed in the mapping view and the corresponding value stored within
+the markdown string. For instance, the username for an `@mention` is contained in the `text` attribute, which is used for display in the mention
+view, while the user ID is contained in the `key` attribute which is used to store within the markdown string.
+
+_Component Name_
+
+-   `nimble-mapping-mention-user`
+
+_Props/Attrs_
+
+-   `key`: string | number | boolean
```

Comment:
To expand on this, it looks like the API is being written this way because of the base class. The Mapping base class is abstract, I think we should just make the key type generic and then have the different implementations fulfill the type as needed, i.e.:

![image](https://github.com/ni/nimble/assets/1588923/e351a2d2-8375-4521-a645-734f119ae8b2)

And

![image](https://github.com/ni/nimble/assets/1588923/5d2e4d59-9202-42f1-8b58-f4d8ee54ce02)

And the rich text mapping elements can just do string if that's the most reasonable.

Reply:
Yeah, the only reason I added all the types is because of the mapping base class but with the above approach, we can pass whatever type we wanted. I have removed other types of `key` and added only `string`